### DISCUSSION TITLE
Splink modules not correctly installing

### DISCUSSION THREAD

**Discussion by bgblackley1** [moj-analytical-services/splink#2415] [Q&A]
I have installed splink 4.0.1 recently but not all the modules are showing up, when I try and import some modules like Linker, block_on, and internals, I get a module not found error. I can import splink itself and the instillation is complete and pointing to the right location it is just missing some of the modules. I printed all the modules in splink and ones I'm trying to import don't show up, so it isn't installing everything. I have tried multiple new enviroments and reinstalling however the issue still persists.

Any help would be greatly appreciated,
Thanks

**Comment by RobinL** [comment]
[EDIT] - I've just experienced some problems on our CI that may be related.  I recommend updating `pip` or `poetry` or whatever tools you are using to manage your environment.

If you still having problems we'd need the details requested below

Given that it works correctly on our side, and in Colab on the examples, I'm fairly confident this is an issue on your side.  Do you have a reproducible example of the commands your using to set up your environment and the Splink script you're attempting to run?  It'd also be useful to know what system type you're running on (windows Linux etc)

**Comment by HappyElon** [comment]
I have encountered the same issue with comparison_template_library.

I want to create an app that will link together 2 datasets and remove or highlight same records.
But when I try to run test sample from here: https://moj-analytical-services.github.io/splink/topic_guides/splink_fundamentals/settings.html
I receive this error:

![изображение](https://github.com/user-attachments/assets/5d42e6cd-6fe8-477f-9a1d-7e85300d4cf2)

I also tried to run this sample in PyCharm but error also appears here:

![изображение](https://github.com/user-attachments/assets/71ae336a-5828-4bed-bcb8-71c7cf1581ef)
![изображение](https://github.com/user-attachments/assets/d180fe74-fba4-44db-865c-3b1c4ccf41c1)

Seems like there is outage of package.

Thanks in advance for your help.

**Reply by RobinL** [reply]
Do you still get the error after upgrading pip/poetry or whatever you're using for environment management?

**Reply by HappyElon** [reply]
I upgraded it via cmd 
![изображение](https://github.com/user-attachments/assets/5cfbfff7-3229-40e6-93fa-8f4e79e4895f)

And also upgraded pip package in PyCharm
![изображение](https://github.com/user-attachments/assets/cb5a1006-447b-48fc-9712-b800ef186616)

It's the latest version for now according to pypi
![изображение](https://github.com/user-attachments/assets/8360522c-259f-4812-8ad2-140433ede73b)

And there's still a problem with package

**Reply by HappyElon** [reply]
I also tried to import it in Google Colab and there's also a problem with comparison_template_library. 

![изображение](https://github.com/user-attachments/assets/3e9a98e1-439e-4cd3-a68d-a3dd9a20f414)
![изображение](https://github.com/user-attachments/assets/507c2586-e3c5-4c8e-a10e-4b4c29c6fedb)

**Reply by RobinL** [reply]
`comparison_template_library` was dropped from Splink v4.  I've checked and the [example notebooks](https://moj-analytical-services.github.io/splink/demos/examples/examples_index.html) seem to be running fine in Colab.

But that gives me a hint at the source of the issue.  [Other people have reported](https://github.com/moj-analytical-services/splink/discussions/2316#discussioncomment-10357142) that if you update Splink from 3 to 4 within an existing virtual environment it can cause problems.  The solution is to delete the venv and recreate it from scratch.

**Reply by HappyElon** [reply]
If there's no comparison_template_library in Splink v4 that means my problem is solved :)

This part in user guide misdirected me, maybe it should be redacted. 
![изображение](https://github.com/user-attachments/assets/f1fff9a4-dbfc-4beb-beaf-f3f7ccaa7d2c)

As I see in samples CTL's features were added to CL and CLL, so I should use them.

Thanks for your help!

**Reply by RobinL** [reply]
Sorry for the trouble and the intiial misdirected solution! Thanks for the heads up we will fix!    (Or if you want credit feel free to PR the file:  https://github.com/moj-analytical-services/splink/edit/master/docs/topic_guides/splink_fundamentals/settings.md. )

**Comment by mrbenhop** [comment]
The worked Example on this page (below) still contains an example that uses `comparison_template_library` too. Has it been left there by mistake or is there a reason to leave older examples in? 

Genuine question as I'm completely new to github and any opensource python stuff. Would it be customary for other contributors etc to update it? And if so, how would they become a collaborator / contributor? I guess there's some sort of process... guess I should ask Chat GPT :D 

Here's the link to the page

https://moj-analytical-services.github.io/splink/charts/comparison_viewer_dashboard.html__

**Reply by RobinL** [reply]
Thanks! 
You're right - it's been left there by mistake from an older version of Splink.  Yes you're welcome to put in a pull request to fix if you'd like to!  Chat GPT should be able to talk you through the steps if you want to.  No pressure though, we'll try to remember to update at some point if not.
